src/rail_gui.cpp

```diff
@@ -1200,21 +1196,9 @@ struct BuildRailStationWindow : public PickerWindowBase {
 				if (!_railstation.newstations) {
 					size->width = 0;
 					size->height = 0;
-					break;
-				}
-
-				/* If newstations exist, compute the non-zero minimal size. */
-				Dimension d = {0, 0};
-				StringID str = this->GetWidget<NWidgetCore>(widget)->widget_data;
-				for (auto station_class : this->station_classes) {
-					StationClass *stclass = StationClass::Get(station_class);
-					for (uint j = 0; j < stclass->GetSpecCount(); j++) {
-						const StationSpec *statspec = stclass->GetSpec(j);
-						SetDParam(0, (statspec != nullptr && statspec->name != 0) ? statspec->name : STR_STATION_CLASS_DFLT_STATION);
-						d = maxdim(d, GetStringBoundingBox(str));
-					}
+				} else {
+					size->width = 0;
```

Comment:
Don't both if-branches have `size->width = 0;` now? So, can't it be moved out of the if?